featurebyte/config.py

```diff
@@ -271,20 +271,58 @@ class Configurations:
         skipped_members=["get_client"],
     )
 
-    def __init__(self, config_file_path: Optional[str] = None) -> None:
+    _instance = None
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> "Configurations":
+        if Configurations._instance is None:
```

Comment:
there's a singleton meta class that could potentially be used i think?

https://github.com/featurebyte/featurebyte/blob/main/featurebyte/common/singleton.py

Reply:
I just checked. It seems like the flow when not init is okay. But when it is initialized, it hits the `SingletonMeta::__call__` and ends there.
```python
def __call__(cls, *args: Any, **kwargs: Any) -> Any:
    if cls not in cls._instances:
        instance = super().__call__(*args, **kwargs)
        cls._instances[cls] = instance
    return cls._instances[cls]
```

However, the early return will mean that I am unable to reinitalize a different config file if a user specifies another config file to use.

Reply:
That is to say, 
```python
# This is init-ed to ~/.featurebyte/config.yaml
config = Configuration()
# /asd.yaml will never be read because the SingletonMeta will just return the already initialized instance
config2 = Configuration("/asd.yaml")  
```

Reply:
for motivation, see description edit